AsaApi/Core/Private/Ark/HooksImpl.cpp

```diff
@@ -165,11 +167,66 @@ namespace AsaApi
 		AGameState_DefaultTimer_original(_this);
 	}
 
-	void Hook_AShooterGameMode_BeginPlay(AShooterGameMode* _AShooterGameMode)
-	{
-		AShooterGameMode_BeginPlay_original(_AShooterGameMode);
-		dynamic_cast<ApiUtils&>(*API::game_api->GetApiUtils()).SetStatus(ServerStatus::Ready);
-	}
+    void Hook_AShooterGameMode_BeginPlay(AShooterGameMode* _AShooterGameMode)
+    {
+        AShooterGameMode_BeginPlay_original(_AShooterGameMode);
+        dynamic_cast<ApiUtils&>(*API::game_api->GetApiUtils()).SetStatus(ServerStatus::Ready);
+        
+		std::uint64_t mask = 0;
+
+		LPWSTR* argv;
+		int argc;
+		int i;
+		FString affinity = "";
+
+		argv = CommandLineToArgvW(GetCommandLineW(), &argc);
+		if (NULL != argv)
+		{
+			for (i = 0; i < argc; i++)
+			{
+				FString arg(argv[i]);
+				FString param = "-affinity=";
+				if (arg.Contains(param))
+				{
+					if (arg.RemoveFromStart(param))
+					{
+						affinity = arg;
+						break;
+					}
+				}
+			}
+
+			LocalFree(argv);
+		}
+
+		if (affinity.IsEmpty())
+			return;
+
+		TArray<FString> parsed;
+		affinity.ParseIntoArray(parsed, L",", true);
+		for (const FString& token : parsed)
+		{
+			try
+			{
+				int coreIndex = std::stoi(token.ToString());
+
+				if (coreIndex >= 0 && coreIndex < 64)
+					mask |= (1ULL << coreIndex);
+				else
+					Log::GetLog()->warn("Warning: core index {} is out of valid range 0..63", coreIndex);
+			}
+			catch (const std::exception&)
+			{
+				Log::GetLog()->warn("Warning: invalid core index \"{}\"", token.ToString());
+			}
+		}
+
+		if (mask != 0)
+		{
+			SetProcessAffinityMask(GetCurrentProcess(), mask);
```

Comment:
This is OK but we should really resolve this via the loader.

Reply:
The problem I had doing this the loader is the affinity would get reverted back to all cores when set before the server is loaded fully.